### DISCUSSION TITLE
Unable to generate payload_freertos_bm.bin

### DISCUSSION THREAD

**Discussion by gege0500** [polarfire-soc/polarfire-soc-documentation#234] [Applications and demos] [answered]
To build a FreeRTOS + BM AMP configuration:
Build the mpfs-rpmsg-freertos project in Master build mode configuration
Build the mpfs-rpmsg-bm project in Remote build mode configuration
Generate a payload using the reference hss-payload-freertos_bm.yaml available in the resources directory:
    $ cd polarfire-soc-amp-examples/resources
    $ hss-payload-generator -c hss-payload-freertos_bm.yaml payload_freertos_bm.bin

cmd error:
In STATE_NEW_PAYLOAD : illegal token >>skip-opensbi<< (0)
please check configuration file

**Comment by hughbreslin** [comment, marked as answer]
Hi @gege0500 what version / release of the HSS payload generator are you using?

**Reply by gege0500** [reply]
v0.99.12

**Reply by hughbreslin** [reply]
Ahhh ok - the "skip-opensbi" flag was added in the [v0.99.24](https://github.com/polarfire-soc/hart-software-services/commit/d2274594b0bf9dfe4987bea4a5bdefc58a111295#diff-1dc19fd57444583769c58441cd442eec1229cc5b991b7bd140d1936d54407baf) increment.

Can you try updating to the latest version and try that out? Theres a pre-built binary available with the [v2022.09](https://github.com/polarfire-soc/hart-software-services/releases/tag/v2022.09) tag.
